core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/auth/BaseGroupsRule.scala

```diff
@@ -280,13 +281,20 @@ abstract class BaseGroupsRule(val settings: Settings)
     }
   }
 
-  private def resolveGroupIds[B <: BlockContext](blockContext: B) = {
-    settings.permittedGroupIds.resolveGroupIds(blockContext)
+  private def resolveGroupsLogic[B <: BlockContext](blockContext: B) = {
+    settings.resolvableGroupsLogic.resolve(blockContext)
   }
 }
 
 object BaseGroupsRule {
 
-  final case class Settings(permittedGroupIds: ResolvablePermittedGroupIds,
-                            usersDefinitions: NonEmptyList[UserDef])
+  final case class Settings[LOGIC <: ResolvableGroupsLogic](resolvableGroupsLogic: LOGIC,
```

Comment:
Why do you think we should stop calling it the `permittedGroup` (I'm talking about the param name)?

I see the following reasons to don't do that:
1. there is no need to add info about the resolvability because the intention is expressed with type
2. in this file, when we talk about groups, we can have in mind:
  * permitted/allowed groups - the ones configured in the settings
  * available groups - the ones that the rule calculates to be available in the tenancy selector on the ROR KBN side
  * local groups - the ones defined in ROR settings
  * external groups - the ones that come from the external services (like LDAP)
So, saying, "here is a group logic", the next, natural question might be: "Which ones?" ;)